code/controllers/subsystem/explosions.dm

```diff
@@ -192,28 +288,34 @@ GLOBAL_LIST_EMPTY(explosions)
 		else
 			dist = EXPLODE_NONE
 
-		//------- EX_ACT AND TURF FIRES -------
-
 		if(T == epicenter) // Ensures explosives detonating from bags trigger other explosives in that bag
 			var/list/items = list()
 			for(var/I in T)
 				var/atom/A = I
-				if (!(A.flags_1 & PREVENT_CONTENTS_EXPLOSION_1))
-					items += A.GetAllContents(ignore_flag_1 = PREVENT_CONTENTS_EXPLOSION_1)
+				if (A.contents && A.contents.len && !(A.flags_1 & PREVENT_CONTENTS_EXPLOSION_1)) //The atom/contents_explosion() proc returns null if the contents ex_acting has been handled by the atom, and TRUE if it hasn't.
```

Comment:
```suggestion
				if (length(A.contents) && !(A.flags_1 & PREVENT_CONTENTS_EXPLOSION_1)) //The atom/contents_explosion() proc returns null if the contents ex_acting has been handled by the atom, and TRUE if it hasn't.
```

Reply:
is this not a proc call and therefore additional overhead?